torch/optim/lr_scheduler.py

```diff
@@ -154,8 +154,12 @@ def __init__(self, optimizer, step_size, gamma=0.1, last_epoch=-1):
         super(StepLR, self).__init__(optimizer, last_epoch)
 
     def get_lr(self):
-        return [base_lr * self.gamma ** (self.last_epoch // self.step_size)
-                for base_lr in self.base_lrs]
+        if (self.last_epoch == 0) or (self.last_epoch % self.step_size != 0):
+            return [group['lr'] for group in self.optimizer.param_groups]
+        return [group['lr'] * self.gamma
+                for group in self.optimizer.param_groups]
+        # return [base_lr * self.gamma ** (self.last_epoch // self.step_size)
+        #         for base_lr in self.base_lrs]
```

Comment:
nit: I can see why you want to leave the old version of the code commented out (the non-base version is harder to understad, because you're defining the transitions, rather than the end state)--it would be even better if you explicitly added a comment saying so (otherwise, the reader will be puzzled about what this mysterious commented out code is. Did someone forget to delete? Is it an alternate implementation?)

Reply:
I am sorry I actually forgot to delete these lines. I think a better way is to delete the old implementation altogether, while highlighting the change in logic in the description of each function. Do you agree?